app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,38 @@
+class CommentsController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    @comments = Comment.all
+  end
+
+def create
+   @comment = Comment.new(comment_params)
+ if @comment.save
+    @post = @comment.post
+     redirect_to @post 
+    flash.notice = "Comment added"
```

Comment:
- let's use the relationships here, we have `current_user` already why send it. 
`@comment = current_user.comments.build(.....)`
- consider using the form errors rails provides here and use render when it fails too